sdk/dotnet/Pulumi.Automation/Serialization/Json/MapToModelJsonConverter.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright 2016-2021, Pulumi Corporation
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Pulumi.Automation.Serialization.Json
+{
+    // necessary because this version of System.Text.Json
+    // can't deserialize a type that doesn't have a parameterless constructor
```

Comment:
Would there be any issues in the context of this package by taking a direct dependency on the latest version of System.Text.Json? The version that shipped with .NET Core 3.1 is pretty well known as being very undercooked.

Reply:
Might need input from the pulumi guys on this since they have a better understanding of how they are using `System.Text.Json` in the core library.

Reply:
We rely on `System.Text.Json` in our core Pulumi package. Why would it be a bigger problem in automation specifically? I think we'll go ahead with this.

Reply:
For clarity's sake, you want to go ahead with referencing a more recent version in the automation API package?

Reply:
Sorry, I actually mean that I'm fine with what you have now